### PR TITLE
chore(deps): update ioredis to v5.0.4

### PR DESCRIPTION
**Pull request by vamsiampolu**
**Changes**

* update ioredis in `peerDependencies` to `5.x`  and `devDependencies` to   `5.0.4`.
* use `ioredis/built/Command` instead of `ioredis/built/command`  to fix tests.
* remove `PromiseContainer` and any references to it.

### CODE REVIEW

**Comment by crobinson42** [comment]
This would be great to get merged and published 🙏

**Comment by crobinson42** [comment]
@stipsan Can I help get a release moved along here? I think this PR is a better solution for ioredis@5 vs #1163 since this removes the promise lib that ioredis no longer supports.

**Comment by stipsan** [comment]
Hi @vamsiampolu and @crobinson42 😄 
I would love to clean house and get rid of that old dusty Promise lib thingy, sure!
All I need is a new section under [Breaking Changes](https://github.com/stipsan/ioredis-mock#breaking-changes) and let's get this PR merged 😌

**Review by stipsan — APPROVED** [review]
Hey I merged https://github.com/stipsan/ioredis-mock/pull/1166 first as it didn't require any changes to the README to land.
Everything else in the PR looks good 👍

File: README.md [hunk 1]
```diff
@@ -54,6 +54,13 @@ console.log(await redis.get('foo'))
 
 ## Breaking Changes
 
+### from v7 to v8
```
Comment (vamsiampolu) [inline README.md:57]: @stipsan  Updated breaking changes in the readme.

**Comment by vamsiampolu** [comment]
Is there something I can do to help with `remote: Permission to vamsiampolu/ioredis-mock.git denied to github-actions[bot].`

**Comment by stipsan** [comment]
> Is there something I can do to help with `remote: Permission to vamsiampolu/ioredis-mock.git denied to github-actions[bot].`

No, that task isn't supposed to run on forks and I've tried to make it only run on upstream multiple times but it's stubborn D:

Thanks for the bonus fixes you pushed to the e2e suite btw ❤️  I'll merge this and fix the build errors on main!

**Pull request merged**